tests/testthat/test-merge_sce_list.R

```diff
@@ -298,9 +302,154 @@ test_that("merging SCEs without names works as expected", {
   )
 })
 
-test_that("merging SCEs with library metadata fails as expected", {
+test_that("merging SCEs with library metadata fails as expected, no altexps", {
   # add library metadata to one of the objects in the list
   metadata(sce_list$sce1)$library_metadata <- "library_metadata"
 
-  expect_error(merge_sce_list(sce_list))
+  expect_error(
+    merge_sce_list(
+      sce_list
+      )
+  )
+})
+
+
+# Tests with altexps -------------------------------------------------------
+
+
+## helper function to add an altExp to a simulated SCE ----
+add_sce_altexp <- function(
+    sce,
+    batch,
+    altexp_name,
+    num_altexp_features,
+    n_cells) {
+
+  sce_alt <- sim_sce(
+    n_genes = num_altexp_features,
+    n_cells = n_cells,
+    n_empty = 0)
+
+  # ensure matching barcodes
+  colnames(sce_alt) <- colnames(sce)
+
+  # add some rowdata columns
+  rowData(sce_alt)[["feature_column"]] <- rownames(sce_alt)
+  rowData(sce_alt)[["other_column"]] <- runif(nrow(sce_alt))
+
+  # add a coldata columns
+  colData(sce_alt)[["coldata_column"]] <- runif(ncol(sce_alt))
+
+  # add logcounts
+  logcounts(sce_alt) <- counts(sce_alt)
+
+  # add metadata
+  library_id <- glue::glue("library-{batch}")
+  sample_id <- glue::glue("sample-{batch}")
+
+  metadata(sce_alt)$library_id <- library_id
+  metadata(sce_alt)$sample_id <- sample_id
+  metadata(sce_alt)$mapped_reads <- 100
+
+  # add sce_alt as sce's altExp, named "adt"
+  altExp(sce, "adt") <- sce_alt
```

Comment:
Since you added the arg, you probably wanted to use it!
```suggestion
  # add sce_alt as sce's altExp
  altExp(sce, altexp_name) <- sce_alt
```

Reply:
🙄